src/transformers/models/llama/tokenization_llama_fast.py

```diff
@@ -143,8 +143,8 @@ def update_post_processor(self):
         eos = self.eos_token
         eos_token_id = self.eos_token_id
 
-        single = f"{(bos+':0 ') * self.add_bos_token}$A:0{(' '+eos+':0') * self.add_eos_token}"
-        pair = f"{single}{(' '+bos+':1') * self.add_bos_token} $B:1{(' '+eos+':1') * self.add_eos_token}"
+        single = f"{(bos+':0 ') * self.add_bos_token}$A:0{(' '+eos+':0') if self.add_eos_token else ''}"
+        pair = f"{single}{(' '+bos+':1') * self.add_bos_token} $B:1{(' '+eos+':1') if self.add_eos_token else ''}"
```

Comment:
I don't think we can just change this in the tokenizer - it's going to change the output and behaviour for llama users

Reply:
It should not, I think this is a bug fix (should be). The llama tests don't make sure that if eos = None this still works. In this case it was failing